lib/rubocop/cop/rspec/repeated_subject_call.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for repeated calls to subject missing that it is memoized.
+      #
+      # @example
+      #   # bad
+      #   it do
+      #     subject
+      #     expect { subject }.to not_change { A.count }
+      #   end
+      #
+      #   it do
+      #     expect { subject }.to change { A.count }
+      #     expect { subject }.to not_change { A.count }
+      #   end
+      #
+      #   # good
+      #   it do
+      #     expect { my_method }.to change { A.count }
+      #     expect { my_method }.to not_change { A.count }
+      #   end
+      #
+      class RepeatedSubjectCall < Base
+        BLOCK_MSG = 'Calls to subject are memoized, this block is misleading'
+
+        # @!method subject_calls(node)
+        def_node_search :subject_calls, <<~PATTERN
+          (send nil? :subject)
+        PATTERN
+
+        def on_send(node)
+          return unless node.method?(:it)
```

Comment:
We have helpers to detect if a node is an example, ‘example?(node)’ to my best memory. ‘it’ can be a ‘specify’ etc